packages/core/src/babel-plugin-card-schema-transform.ts

```diff
@@ -131,6 +131,9 @@ export default function main(babel: typeof Babel) {
         // @babel/plugin-proposal-class-properties creates a "Class" visitor
         // that you need to hook into.
         for (let bodyItem of path.get('body').get('body')) {
+          if (type === 'primitive') {
+            throw error(bodyItem, `primitive cards cannot have fields`);
+          }
```

Comment:
we should perform this check before we try to transpile the schema, and the error that we throw should describe the problem. i think packages/core/src/compiler.getParentCard() is probably the best place to check for this scenario and throw a meaningful error